src/benchmark.cpp

```diff
@@ -5,69 +5,248 @@
+        case CUSTOM_SVG_WITH_QR_BR:
+            return PARSER_CUSTOM_MARKER;
+        case ARUCO_WITH_QR_BR:
+        case TWO_ARUCO_WITH_QR_BR:
+            return PARSER_ARUCO;
+        case SQUARES_WITH_QR_BR:
+        case SQUARE_OUTLINES_WITH_QR_BR:
+        default:
+            return PARSER_DEFAULT;
+    }
 }
 
 int main(int argc, char* argv[]) {
-    if (argc != 4) {
-        fprintf(stderr, "usage: parser OUTPUT_DIR ATOMIC_BOXES DIR_IMAGE...\n");
+    std::string output_dir_opt = "./output";
+    std::string atomic_boxes_file = "./original_boxes.json";
+    std::string input_dir = "./copies";
+    std::string copies_str = "1";
+    int encoded_marker_size = 15;
+    int fiducial_marker_size = 10;
+    int grey_level = 0;
+
+    std::string input;
+
+    // Affichage de la bannière
+    display_banner();
+
+    // Récupération des entrées utilisateur avec un format amélioré
+    output_dir_opt = get_user_input("Output directory", output_dir_opt);
+    atomic_boxes_file = get_user_input("Atomic boxes JSON file path", atomic_boxes_file);
+    input_dir = get_user_input("Input directory", input_dir);
+    copies_str = get_user_input("Number of copies", copies_str);
+    encoded_marker_size = get_user_input_int("Encoded marker size", encoded_marker_size, 5, 50);
+    fiducial_marker_size = get_user_input_int("Fiducial marker size", fiducial_marker_size, 5, 50);
+    grey_level = get_user_input_int("Grey level", grey_level, 0, 255);
+    int marker_config_default = display_configs(marker_configs, ARUCO_WITH_QR_BR, "Available marker configurations:");
+    int marker_config = get_user_input_int("Marker configuration (1-10)", marker_config_default, 1, 10);
+
+    // Sélectionne le parseur à utiliser en fonction de la configuration choisie
+    std::string selected_parser = select_parser_for_marker_config(marker_config);
+
+    // Affichage du récapitulatif
+    std::cout << std::endl << BOLD << "Configuration:" << RESET << std::endl;
+    std::cout << "- Output directory: " << YELLOW << output_dir_opt << RESET << std::endl;
+    std::cout << "- Atomic boxes file: " << YELLOW << atomic_boxes_file << RESET << std::endl;
+    std::cout << "- Input directory: " << YELLOW << input_dir << RESET << std::endl;
+    std::cout << "- Copies: " << YELLOW << copies_str << RESET << std::endl;
+    std::cout << "- Encoded marker size: " << YELLOW << encoded_marker_size << RESET << std::endl;
+    std::cout << "- Fiducial marker size: " << YELLOW << fiducial_marker_size << RESET << std::endl;
+    std::cout << "- Grey level: " << YELLOW << grey_level << RESET << std::endl;
+    std::cout << "- Marker config: " << YELLOW << marker_config << RESET << std::endl << std::endl;
+
+    // Verify values are not empty (should never happen with defaults)
+    if (output_dir_opt.empty() || atomic_boxes_file.empty() || input_dir.empty() || copies_str.empty()) {
+        std::cerr << "Required arguments cannot be empty.\n";
         return 1;
```

Comment:
```suggestion
        throw std::runtime_error("Required arguments cannot be empty.");
```
peut-être faire un thow ?